test/unix/main.ml

```diff
@@ -143,23 +238,86 @@ let readonly () =
               (Printf.sprintf "Wrong insertion: %s value is missing." v))
     !l
 
+let readonly_no_flush () =
+  let w =
+    Index.v ~fresh:true ~readonly:false ~log_size ~fan_out_size index_name
+  in
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  List.iter (fun (k, v) -> Index.add w k v) !l;
+  test_find_present r
+
+let readonly_lwt () =
+  let w =
+    Index.v ~fresh:true ~readonly:false ~log_size ~fan_out_size index_name
+  in
+  let r =
+    Index.v ~shared:false ~fresh:false ~readonly:true ~log_size ~fan_out_size
+      index_name
+  in
+  let wp = Lwt_list.iter_p (fun (k, v) -> Lwt.return (Index.add w k v)) !l in
+  let p =
+    Lwt.bind
+      (Lwt.bind wp (fun _ -> Lwt.return (Index.flush w)))
+      (fun _ -> test_find_present_lwt r)
+  in
+  Lwt_main.run p
```

Comment:
If you use `Alcotest_lwt.test_case` instead, you can avoid a call to `Lwt_main.run` here (since that `test_case` expects a `unit Lwt.t`)